osquery/events/linux/probes/probes.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/events/linux/probes/probes.h>
+
+#include <osquery/utils/expected/expected.h>
+#include <osquery/utils/map_take.h>
+#include <osquery/utils/system/linux/perf_event/perf_event.h>
+
+#include <osquery/logger.h>
+
+#include <sys/ioctl.h>
+
+namespace osquery {
+namespace events {
+
+namespace {
+
+Expected<std::string, LinuxProbesControl::Error> toTracingPath(
+    syscall::Type type) {
+  static const auto table =
+      std::unordered_map<syscall::Type, std::string, EnumClassHash>{
+          {syscall::Type::KillEnter, "syscalls/sys_enter_kill"},
+          {syscall::Type::KillExit, "syscalls/sys_exit_kill"},
+          {syscall::Type::SetuidEnter, "syscalls/sys_enter_setuid"},
+          {syscall::Type::SetuidExit, "syscalls/sys_exit_setuid"},
+      };
+  auto exp = tryTakeCopy(table, type);
+  if (exp.isError()) {
+    return createError(LinuxProbesControl::Error::InvalidArgument,
+                       "unknown tracing event path for type ",
+                       exp.takeError())
+           << to<std::string>(type);
+  }
+  return exp.take();
+}
+
+Expected<EbpfTracepoint, LinuxProbesControl::Error> createTracepointForSyscall(
+    syscall::Type type, PerfEventCpuMap const& cpu_map) {
+  auto program_exp = genLinuxProgram(BPF_PROG_TYPE_TRACEPOINT, cpu_map, type);
+  if (program_exp.isError()) {
+    return createError(LinuxProbesControl::Error::SystemEbpf,
+                       "could not load program to track syscall ",
+                       program_exp.takeError())
+           << to<std::string>(type);
+  }
+  auto tracing_path_exp = toTracingPath(type);
```

Comment:
Should this be ```const```?

Reply:
Thank you for asking!
It could not, because later the value from expected is moved out by `take()` method, which is not `const` by design.